### ISSUE TITLE
Feature Request: Expose AV1 and allow download & playback

### ISSUE THREAD

**Issue by ich0x** [TeamNewPipe/NewPipe#2849]
I have personally tested AV1 playback on an old 2016 Samsung J3 running LOS 14.1 and was successful in finding and playing suitable 480p AV1 videos on youtube using youtube-dl & a dev version of VLC from artifacts.videolan.org/vlc-android/

I have also tested 720p AV1 on a newer 2018 model J4. 

However NewPipe completely ignores this format. 
I would like to see this format added to NewPipe, incorporating dav1d decoding somehow into exoplayer

**Comment by Redirion** [comment]
[ExoPlayer 2.11 will support AV1 through libgav1](https://medium.com/google-exoplayer/playing-av1-videos-with-exoplayer-a7cb19bedef9). But if I recall correctly NewPipe didn't want to include native code dependencies. So it might never get included.

**Comment by cool-student** [comment]
> Android 10 includes an AV1 decoder. This means ExoPlayer already supports AV1 videos on devices running Android 10, but until now, there was no support for devices running older Android versions.

This is kinda cool so on Android 10(+) no extra libraries are needed.

**Comment by Redirion** [comment]
this also requires an Extractor change @Stypox

**Comment by fabiorug** [comment]
Do you plan to add dav1d 0.7.1 in Newpipe?

**Comment by wb9688** [comment]
@fabiorug: No, ExoPlayer doesn't support that. It does have a `libgav1` extension though, but we don't want to add native code to NewPipe. That means we won't support it as long as it isn't a built-in feature of Android. We could however add downloading of AV1 videos, if you could provide a link to any video on a service that's currently supported by NewPipe that is encoded by AV1.

**Comment by MD77MD** [comment]
>This is kinda cool so on Android 10(+) no extra libraries are needed.
Is it true that in android 10+, AV1 is built-in.

**Comment by wb9688** [comment]
@MD77MD: https://developer.android.com/guide/topics/media/media-formats#video-codecs doesn't mention anything about AV1 being supported by Android in any version, however https://developer.android.com/about/versions/10/highlights#new_audio_and_video_codecs says it has indeed been added in Android 10.

**Comment by opusforlife2** [comment]
The first page does show AV1 in the table. Maybe it was update in the meanwhile.

**Comment by wb9688** [comment]
@opusforlife2: Yup, seems like they finally updated it. Anyway could someone link a YouTube video that has an AV1 stream (according to `youtube-dl -F`)?

**Comment by opusforlife2** [comment]
Dude, Youtube has entire playlists to test AV1. Give me a second.

**Comment by opusforlife2** [comment]
Yup. This one. https://www.youtube.com/playlist?list=PLyqf6gJt7KuHBmeVzZteZUlNUQAVLwrZS

**Comment by Snowden-rgb** [comment]
lol i am in for this feature

**Comment by opusforlife2** [comment]
Pffft. I am in for this feature as well.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
+1

**Comment by D4v1dH03** [comment]
AV1 is now out of beta and officially adopted in nearly all videos on YouTube, any plans to support AV1 in Newpipe?

**Comment by opusforlife2** [comment]
@D4v1dH03 See: https://github.com/TeamNewPipe/NewPipeExtractor/pull/706.

BTW, do you have a source for that? Was there a blog post or something?

**Comment by D4v1dH03** [comment]
> @D4v1dH03 See: https://github.com/TeamNewPipe/NewPipeExtractor/pull/706.
> 
> BTW, do you have a source for that? Was there a blog post or something?

I tested a NCS music playlist with `youtube-dl -F` and discovered that they all have AV1 encoded formats available  


One of the output shows this
```
399          mp4        1920x1080  1080p60 3991k , mp4_dash container, av01.0.09M.08@3991k, 60fps, video only, 102.29MiB
```

**Comment by opusforlife2** [comment]
Oh. I've seen that. I thought you meant Google officially declared AV1 out of beta on Youtube or something.

Though just the fact that it's now being used for popular videos means it's probably considered stable.

**Comment by utack** [comment]
At least the offical Youtube app does stream it when supported in hardware on phones, I guess they do consider it the superior format for that use case

**Comment by opusforlife2** [comment]
@utack Have you seen that on Mediatek Dimensity phones? As far as I know, those are the only chips that support AV1 hardware decoding right now.

**Comment by utack** [comment]
@opusforlife2 I have seen it on my Galaxy S8, the european Exynos edition has a 4k60 capable AV1 decoder  

And on my Samsung Tizen "The Frame" TV as well, but I am not sure if that app is considered offical

**Comment by opusforlife2** [comment]
Ah. Exynos 2100 added AV1 decode support. Good to know.

**Comment by D4v1dH03** [comment]
Many phones don't have hw decoder unfortunately but at least they have sw decoder

**Comment by BurhanBudak** [comment]
Bump

Firefox Mobile uses software decoding and it loads faster then Newpipe, official app and the delisted Vanced.

**Comment by Plugdemholes-Imeanpatchdembugs** [comment]
I too would be interested, just because the higher pixel count of the tablets makes it easier to see some difference in the encoding result :P

**Comment by DavidH-Tech** [comment]
Android 14 will make AV1 mandatory even if no hardware support, the system will support AV1

**Comment by Crystal-RainSlide** [comment]
Note: Starting from 2023, mid-level phones and beyond should be mostly capable of decoding (at least) 720p AV1 video, both hw and sw decoding.

https://gist.github.com/nekotrix/82158fc48f7edb0db8338779a997dad0

And I'm pretty sure "Not Qualcomm ones." is not true.

**Comment by n-ce** [comment]
The Soc on my phone does not support AV1 so I downloaded a video player to play an AV1 video file via software decoding, to my surprise it was running on hardware decoding. Later found out Android 14 supports it natively so I hope this will be taken seriously now for the compose refactor.

**Comment by robzorua** [comment]
There's no need to support VLC's libdav1d anymore because Google/YouTube will force it to Android even if you're using entry level, they will change their libgav1 to libdav1d.

https://www.androidauthority.com/android-update-av1-videos-3420418/

The only problem is why they put AV1 to MP4 not with WEBM/OPUS? WEBM/OPUS/AV1 will give better compression I hope they'll re-encode all their videos to WEBM/AV1.

**Comment by n-ce** [comment]
> The only problem is why they put AV1 to MP4 not with WEBM/OPUS? WEBM/OPUS/AV1 will give better compression I hope they'll re-encode all their videos to WEBM/AV1.

MP4 / WEBM are just containers it doesn't matter much. VP9 didn't support the MP4 container format so I'm guessing they did this to generalize the AV1 codec among people. Btw I [download AV1(+Opus) videos via yt-dlp](https://gist.github.com/n-ce/b232e9f4457b96a30f16b50bdfc8402d) all the time and they are all contained by WEBM.

**Comment by Ciancy28** [comment]
> VP9 didn't support the MP4 container format

Actually it did, and Google Photos as an example, uses vp9+aac in mp4 container when downloading videos that were uploaded in storage saver mode.
> Btw I [download AV1(+Opus) videos via yt-dlp](https://gist.github.com/n-ce/b232e9f4457b96a30f16b50bdfc8402d) all the time and they are all contained by WEBM.

That's because yt-dlp automatically remuxes stuff when it makes sense, since opus audio is already webm and av1 can be contained in webm, it decides to remux av1 in the same webm container, downloading a single av1 format without audio gives you av1 in mp4 container (as expected, since av01 is listed as mp4 when looking at the formats list)
yt-dlp also remuxes in .mkv when downloading opus+h264, but of course youtube doesn't offer such format natively.
> are just containers it doesn't matter much

this part is correct, there's a negligeable amount of overhead using mp4 instead of webm but it is only measurable on very short low quality videos (and even there, we are talking few kilobytes).

**Comment by Plugdemholes-Imeanpatchdembugs** [comment]
Since the end of 2022, half year after I posted, I finally got a graphics card with AV1 decoding capability, I also added enhanced-h264ify which has the option to disable h264, VP8, VP9 or AV1.
I now have always VP8 and VP9 disabled, so I'm usually limited to max. 1080p for h264 but when a AV1 video plays you notice the difference instantly even though my standard playback resolution is only 1080p. But to my regret I have to say YT basicly only seems to encode videos that get uploaded with at least 1440p as AV1 and by my own observation I'd say I'm lucky when every 25th video I'm watching got encoded as AV1.
So I currently don't see a trend that YT wants to increase AV1 encoded content to sub 1440p videos... maybe we are more lucky by the end of 2027.

**Comment by n-ce** [comment]
> But to my regret I have to say YT basicly only seems to encode videos that get uploaded with at least 1440p as AV1. 

That is not true YT now encodes a variety of videos in AV1, they just don't give it to all viewers. 4/5 videos nowadays are encoded in AV1. The rest of them either don't have enough views or is not satisfying an unknown metric that I don't understand yet.
To test the 4/5 claim, just switch to 240p/144p.

**Comment by Plugdemholes-Imeanpatchdembugs** [comment]
Can only speak to what I've personally observed since end of 2022. Didn't expect 240p/144p is still in demand anyways.

**Comment by PsykoRider** [comment]
@theScrabi 

Will the AV1 codec be added to NewPipe at some point? I see that this thread was created years ago, yet NewPipe still hasn’t implemented this codec for playback and downloading. Best regards.

**Comment by ich0x** [comment]
The PipePipe fork added this awhile ago, so the code already exists
somewhere at least.

**Comment by Plugdemholes-Imeanpatchdembugs** [comment]
Didn't know about Pipepipe should probably use it in the future if I want to download videos in AV01, eh?
So basically the WebM container in Newpipe still only delivers the VP9 variant did I get this right?

Was kinda smitten to see that in Pipepipe I can select Av01, VP9 or even AVC1 for streaming and downloading.
I wanted to download the 1440p(or 4K) versions and higher for half a year with the new tablet but the size didn't seem worth it, with Pipepipe I can finally get my Av01 downloads flawlessly and the expected reasonable size! ˆˆ

I mean is that going to change any time soon from not beaing able to grab the AV01 with Newpipe. 
I'm not the guy who needs the newest stuff but soon I'll send my old Android device from 2018 into retirement that still has trouble with the WebM 1080p60fps streams would be kinda nice if that would soon be possible to grab the Av01 streams.

**Comment by ich0x** [comment]
Of course the very next day after I mentioned PipePipe, youtube went
nuclear with their new SABR streaming protocol and broke pretty much *all*
the apps I use regarding youtube.  That needs fixing first.
